cou/steps/hypervisor.py

```diff
@@ -198,13 +198,15 @@ def _generate_post_upgrade_plan(self) -> PostUpgradeStep:
         raise NotImplementedError
 
     # pylint: disable=unused-argument
-    def generate_upgrade_plan(self, target: OpenStackRelease) -> UpgradePlan:
+    def generate_upgrade_plan(self, target: OpenStackRelease, force: bool) -> UpgradePlan:
```

Comment:
the planning so far is filtering the hypervisors machines. I think it makes sense accept this parameter in this function. If you think it should be part of another PR I'm fine with it, but we are already adding force so it might be an opportunity.

Reply:
Yes, and with that I need to add `__eq__` to COUMachie, since you cannot do `machine_id in machines`, where machines are `list[COUMachine]`.

Reply:
I'm assuming that won't be part of this PR, so solving it